eng/scripts/generate_from_source_pom.py

```diff
@@ -130,7 +130,19 @@ def create_from_source_pom(project_list: str, set_pipeline_variable: str, set_sk
         fromSourcePom.write(pom_file_end)
 
     if set_pipeline_variable:
-        checkout_paths = list(set(sorted([p.directory_path for p in source_projects])))
+        # The directory_path is too granular. There are build rules for some libraries that
+        # create empty sources/javadocs jars using the README.md. Not every library
+        # has a README.md and, in these cases, it uses the README.md from the root service
+        # directory. This will also trim the number of paths down considerably.
+        service_directories: Set[str] = set()
+        for p in source_projects:
+            # get the service directory, which is one level up from the library's directory
+            service_directory = '/'.join(p.directory_path.split('/')[0:-1])
```

Comment:
Are we sure this correctly finds the service directory in all cases? Should we consider a regex to be sure we find only the paths we want?

Reply:
As an example we do that in some of our shared properties https://github.com/Azure/azure-sdk-for-java/blob/main/eng/common/scripts/Package-Properties.ps1#L148.

Reply:
The directory path is SDK/DirectoryWeActuallyWantToSync/ArtifactDir which is what's being passed off to sync but what we really want is SDK/DirectoryWeActuallyWantToSync. With the way we've enforced the directory structure, this will get us what we want as nothing that we're building will be from a deeper or shallower depth. All I need to do is snip off the subdirectory to get the ServiceDirectory.

Reply:
> As an example we do that in some of our shared properties https://github.com/Azure/azure-sdk-for-java/blob/main/eng/common/scripts/Package-Properties.ps1#L148.

It's worth mentioning that for this script we're expecting everything to be a shipping library however this example wouldn't work for this particular case as it'll only work for things under /SDK. The project list isn't just building/shipping libraries but also includes AdditionalModules which aren't under /SDK but rather /common. If script had been written in powershell I could have just used (Get-Item $file).Directory.Parent on the pom file to get the service directory.